src/eth_provider/provider.rs

```diff
@@ -527,13 +539,20 @@ where
         let chain_id: u64 =
             self.chain_id().await?.unwrap_or_default().try_into().map_err(|_| TransactionError::InvalidChainId)?;
 
-        // Validate the transaction
         let latest_block_header = self.database.latest_header().await?.ok_or(EthApiError::UnknownBlockNumber(None))?;
-        validate_transaction(&transaction_signed, chain_id, &latest_block_header)?;
 
         // Recover the signer from the transaction
         let signer = transaction_signed.recover_signer().ok_or(SignatureError::Recovery)?;
 
+        // Check the cached balance of the signer
+        self.check_cached_balance(signer, transaction_signed.value()).await?;
```

Comment:
this will just take into account the value of the transaction, we need to account for `gas_price * gas_limit`. https://github.com/paradigmxyz/reth/blob/main/crates/transaction-pool/src/traits.rs#L780

Reply:
fixed, thank you